tests/spark/test_spark_model_export.py

```diff
@@ -984,3 +986,60 @@ def test_model_log_with_signature_inference(spark_model_iris, input_example):
     else:
         assert column_names == ["0", "1", "2", "3"]
     assert mlflow_model.signature.outputs == Schema([ColSpec(type=DataType.double)])
+
+
+def test_model_with_vector_input(model_path, spark):
+    from mlflow.types.schema import Array
+
+    mlflow.pyspark.ml.autolog()
+    train_df = spark.createDataFrame(
+        [([3., 4.], 0), ([5., 6.], 1)], schema="features array<double>, label long"
+    ).select(array_to_vector("features").alias("features"), col("label"))
+
+    lor = LogisticRegression(maxIter=2)
+    with mlflow.start_run() as run:
+        lor.fit(train_df)
+
+    model_uri = f"runs:/{run.info.run_id}/model"
+
+    # check vector type is inferred correctly
+    model_info = mlflow.models.get_model_info(model_uri)
+    input_type = model_info.signature.inputs.input_dict()['features'].type
+    assert isinstance(input_type, Array) and input_type.is_sparkml_vector
+
+    assert model_info.signature.outputs.input_dict()['prediction'].type == DataType.double
+
+    model = mlflow.pyfunc.load_model(model_uri)
+
+    test_pdf = pd.DataFrame({'features': [[1., 2.], [3., 4.]]})
+    model.predict(test_pdf)  # ensure enforcing input / output schema passing
+
```

Comment:
Let's add a test where we manually remove the entry `is_sparkml_vector` from a logged signature (to simulate legacy behavior) and ensure that we can load the model as pyfunc and run inference against it with a basic LogReg SparkML model. Better to be safe with this change that is mutating the signature definition for models saved in earlier versions of MLflow.

Reply:
2 cases:
* Saving model with **new version** mlflow (supporting spark vector) and load the saved model by **old version mlflow**,  we are **not** going to support this case, right?
* Saving model with **old version** mlflow and loading it with new version mlflow, this case if the model contains vector type input / output, then logging signature will be **skipped** by old version mlflow because it doesn't support vector type. So do we still need the test https://github.com/mlflow/mlflow/pull/12758#discussion_r1688400596 ?

Reply:
Ah, that's right - we just raise a warning and alert the user that VectorUDT is an unsupported type. Since this isn't a compatibility change and rather just adding feature support for this type, we don't need the test.